packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx

```diff
@@ -156,7 +157,26 @@ export default function ComponentCatalog({ className }: ComponentCatalogProps) {
                 flexDirection={'row'}
                 justifyContent="space-between"
               >
-                <Typography variant="overline">Custom Components</Typography>
+                <Box display="flex" alignItems="center">
+                  <Typography mr={0.5} variant="overline">
+                    Custom Components
+                  </Typography>
+                  <HelpTooltipIcon
+                    helpText={
+                      <Typography variant="inherit">
+                        Expand Toolpad with your own React components.{' '}
+                        <Link
+                          href="https://mui.com/toolpad/concepts/custom-components"
+                          target="_blank"
+                          rel="noopener"
+                        >
+                          Click{' '}
```

Comment:
We need to find a different link label https://developer.chrome.com/en/docs/lighthouse/seo/link-text/

Reply:
@oliviertassinari maybe we can rephrase the help text to make it more descriptive with link 🤔

Reply:
A rephrase would be great. 

It's common to see generic links, but "click" is pushing it too far, "learn more" would already bring a lot more context.

Notion

<img width="343" alt="Screenshot 2023-09-03 at 00 19 00" src="https://github.com/mui/mui-toolpad/assets/3165635/3c6d18d3-6e64-4ba1-ad34-87f157bbbfb9">

Ashby

<img width="371" alt="Screenshot 2023-09-03 at 00 25 04" src="https://github.com/mui/mui-toolpad/assets/3165635/493469de-9b3e-47db-8f02-fc07ddbd4d70">

Reply:
With learn more text :

Light : 
<img width="659" alt="Screenshot 2023-09-04 at 8 35 43 PM" src="https://github.com/mui/mui-toolpad/assets/30728574/0da3a933-8863-44e2-8e3c-ce2b7cfa2d68">

Dark: 
<img width="659" alt="Screenshot 2023-09-04 at 8 35 17 PM" src="https://github.com/mui/mui-toolpad/assets/30728574/0a537a9a-d888-439d-a005-25e92d6a1043">